src/DataFramesMeta.jl

```diff
@@ -378,14 +378,18 @@ end
 ##
 ##############################################################################
 
-function transform(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
-    result = copy(d)
+function transform!(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
     for (k, v) in kwargs
-        result[k] = isa(v, Function) ? v(d) : v
+        d[k] = isa(v, Function) ? v(d) : v
     end
-    return result
+    return d
 end
 
+function transform(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
+    return transform!(copy(d))
+end
+
+# No mutating transform since transform returns a full DataFrame
```

Comment:
As noted at https://github.com/JuliaData/DataFramesMeta.jl/issues/110#issuecomment-425736923, why not mutate the parent?